Cookbook/Technical-Documents/AddingFrameworks.md

```diff
@@ -0,0 +1,76 @@
+
+## Creating a new framework target
+
+If you need to create a new framework target in our project to add a new module, you need to know that we use a `Framework.{debug|release}.xcconfig` configuration file to ensure all our internal frameworks use consistent settings. But since our internal frameworks also use CocoaPods to reference the third-party libraries they use, we need to merge our own `xcconfig` with the ones from CocoaPods.
+
+_The following steps will guide you to configure a new module named `BabylonNewFramework`. Replace every occurrence of `BabylonNewFramework` in the following instructions with the actual name of your new internal framework._
+
+### Add the framework and its dependencies in Podfile
+
+1. Create the framework target (and its associated Test target(s))
+2. Add the framework target (and its test target) to the `Podfile`, listing all its third-party dependencies (like `Bento`, `ReactiveCocoa`, `ReactiveFeedback`, etc) in there
+3. Run `bundle exec pod install` to install the framework's pods
+
+### Add internal dependencies manually
+
+If your framework itself depends on other internal Babylon frameworks (e.g. `BabylonCore`, `BabylonDependencies`, ...), those are not Pods and thus not added via the `Podfile` but are frameworks produced by other projects in our workspace, so you need to add them manually:
+
+1. Add depending internal frameworks to the `"Linked Binary with Libraries"` Build Phase
+   * You only need to do this step if your framework itself depends on other internal frameworks. Avoid any unnecessary dependencies (which also avoids limiting the parallelisation during build)
+   * You need not add dependencies of your dependencies — only the ones that you refer to in code directly
+   * You need only to do that for **internal** frameworks. Third-party frameworks are already linked via CocoaPods thanks to the `Pods-BabylonNewFramework` umbrella framework generated by CocoaPods
+
+
+### Create the merging xcconfig
+
+1. Create a `"BuildConfigs"` folder next to your `BabylonNewFramework.xcodeproj`, so typically in `<RepoRoot>/BabylonNewFramework/BuildConfigs`
+2. Create two new `xcconfig` files in that folder with the following content:
+
+   * `BabylonNewFramework.debug.xcconfig`:
+
+	```
+	#include "../BuildConfigs/Framework.Debug.xcconfig"
+	#include "../Pods/Target Support Files/Pods-BabylonNewFramework/Pods-BabylonNewFramework.debug.xcconfig
+	```
+
+   * `BabylonNewFramework.release.xcconfig`:
+
+	```
+	#include "../BuildConfigs/Framework.Release.xcconfig"
+	#include "../Pods/Target Support Files/Pods-BabylonNewFramework/Pods-BabylonNewFramework.release.xcconfig
+	```
+
+3. Add the `BuildConfigs` folder to your Xcode project (by drag & drop or right-click + "Add Files to ...").
+   * Be extra sure NOT to include the `xcconfig` files in the target when adding the references (the `xcconfig` files should be referenced in the project but should not belong in any target so **uncheck all target checkboxes**)
```

Comment:
> (the `xcconfig` files should be referenced in the project but should not belong in any target so **uncheck all target checkboxes**)

Idea for the projects' configuration - maybe we could `xcconfig` files to `EXCLUDED_SOURCE_FILE_NAMES` build setting to be extra sure about it.

Reply:
Oh... I had the option on my list for Danger rule ideas to make Danger warn about those, but `EXCLUDED_SOURCE_FILE_NAMES` could be a nice solution/alternative indeed!